up42/order.py

```diff
@@ -118,25 +118,33 @@ def place(cls, auth: Auth, order_parameters: dict) -> "Order":
         return order
 
     @staticmethod
-    def estimate(auth: Auth, order_parameters: dict) -> int:
+    def estimate(auth: Auth, order_parameters: dict) -> dict:
         """
         Returns an estimation of the cost of an order.
 
         Args:
             auth: An authentication object.
-            order_parameters: A dictionary like {dataProduct: ..., "params": {"id": ..., "aoi": ...}}
+            order_parameters: A dictionary  like {dataProduct: ..., "params": {"id": ...}, "featureCollection": ...}.
 
         Returns:
-            int: The estimated cost of the order
+            dict: representation of a JSON estimation response with summary, results, and errors.
         """
-        url = endpoint(f"/workspaces/{auth.workspace_id}/orders/estimate")
+        url = endpoint("/v2/orders/estimate")
 
         response_json = auth._request(request_type="POST", url=url, data=order_parameters)
-        estimated_credits: int = response_json["data"]["credits"]  # type: ignore
-        logger.info(
-            f"Order is estimated to cost {estimated_credits} UP42 credits (order_parameters: {order_parameters})"
-        )
-        return estimated_credits
+
```

Comment:
This approach is wrong:
1. We have 400 and 422 responses with completely different signatures and we need to behave differently for those responses
2. If there are no errors, it doesn't make any sense to return them as part of summary
3. What are the cases when we have errors in 200 response? Why do we judge the failure of estimation by totalCredits presence and not by error presence?